packages/core/source-map/src/SourceMap.js

```diff
@@ -0,0 +1,441 @@
+// @flow
+import {SourceMapConsumer, SourceMapGenerator} from 'source-map';
+import lineCounter from '../../utils/src/lineCounter';
+
+type PositionType = {
+  line: number,
+  column: number
+};
+
+type MappingType = {
+  generated: PositionType,
+  original: PositionType | null,
+  source: string | null,
+  name?: string | null
+};
+
+type RawSourceMapType = {
+  version: number,
+  sources: Array<string>,
+  names: Array<string>,
+  sourceRoot?: string,
+  sourcesContent?: Array<string>,
+  mappings: string,
+  file: string
+};
+
+type RawMapInputType = SourceMapConsumer | string | RawSourceMapType;
+
+type ConsumerMappingItemType = {
+  source: string,
+  generatedLine: number,
+  generatedColumn: number,
+  originalLine: number,
+  originalColumn: number,
+  name: string
+};
+
+export default class SourceMap {
+  mappings: Array<MappingType>;
+  sources: Map<string, string | null>;
+  lineCount: number;
+
+  constructor(
+    mappings?: Array<MappingType> = [],
+    sources?: Map<string, string> | {[key: string]: string}
+  ) {
+    // We probably only wanna run this in dev, as sourcemaps should be fast in prod...
+    this._validateMappings(mappings);
+
+    this.mappings = mappings;
+
+    if (sources) {
+      let iteratable =
+        typeof sources === 'object' ? Object.entries(sources) : sources;
+      // $FlowFixMe
+      this.sources = new Map(iteratable);
+    } else {
+      this.sources = new Map();
+    }
+
+    this.lineCount = 0;
+  }
+
+  // This should never be necessary as long as sourcemaps are valid
+  // It's still here in case some asset needs it but should be deprecated over time...
+  filterMappings(mappings: Array<MappingType>): Array<MappingType> {
```

Comment:
Not sure if we should remove this, it's probably no longer necessary, we used to use it in the constructor.
But in theory all mappings should be valid now that we support null mappings properly.